### PR TITLE
Replace missing targets with returns

### PR DESCRIPTION
**Pull request by FWDekker**
Overcomes the bug where the `target` of an `GotoStmt`, `IfStmt`, or `SwitchStmt` has been filtered out in the LUG or the pattern detector while a reference to it remains in another statement, making the body invalid.

### CODE REVIEW

File: modules/mining-pipeline/jimple-evosuite-test-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/testgenerator/jimpleevosuite/ClassGenerator.kt [hunk 1]
```diff
@@ -128,6 +133,29 @@ internal class ClassGenerator(className: String) {
...
+
+    /**
+     * Replaces invalid targets in [statements] with targets to the last statement.
+     *
+     * @param statements the statements to replace invalid targets in
+     */
+    private fun replaceInvalidTargets(statements: List<Unit>) {
+        val targetReplacement = statements.last()
+
+        statements.forEach { stmt ->
```
Comment (casperboone) [inline modules/mining-pipeline/jimple-evosuite-test-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/testgenerator/jimpleevosuite/ClassGenerator.kt]: Please don't abbreviate statement

File: modules/mining-pipeline/jimple-evosuite-test-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/testgenerator/jimpleevosuite/ClassGenerator.kt [hunk 2]
```diff
@@ -128,6 +133,29 @@ internal class ClassGenerator(className: String) {
...
+     * Replaces invalid targets in [statements] with targets to the last statement.
+     *
+     * @param statements the statements to replace invalid targets in
+     */
+    private fun replaceInvalidTargets(statements: List<Unit>) {
+        val targetReplacement = statements.last()
+
+        statements.forEach { statement ->
+            when (statement) {
+                is GotoStmt -> if (!statements.contains(statement.target)) statement.target = targetReplacement
```
Comment (JSAbrahams) [inline modules/mining-pipeline/jimple-evosuite-test-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/testgenerator/jimpleevosuite/ClassGenerator.kt]: Is it not possible to the move the conditional of the if before the arrow?

`is GotoStmt && !statements.contains(statement.target) -> ...`

also, is there a function `doesNotContain`, or something along those lines? I usually prefer those to using `!` as it is easy to read over such a character.

Comment (FWDekker) [reply modules/mining-pipeline/jimple-evosuite-test-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/testgenerator/jimpleevosuite/ClassGenerator.kt]: It is not possible to do
```kotlin
when(statement) {
    is GotoStmt && !statements.contains ...
```
but it is possible to do
```kotlin
when {
    statement is GotoStmt && !statements.contains ...
```

Which one do you prefer?

---

I'm afraid there is no `doesNotContain` or something similar :(

Comment (JSAbrahams) [reply modules/mining-pipeline/jimple-evosuite-test-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/testgenerator/jimpleevosuite/ClassGenerator.kt]: Hm I think I prefer the last one. I read the `->` as a "then do this", and it feels slightly awkward to me to have another conditional (or `if` statement to be exact) right after the arrow itself.

Comment (FWDekker) [reply modules/mining-pipeline/jimple-evosuite-test-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/testgenerator/jimpleevosuite/ClassGenerator.kt]: @JSAbrahams Done! What do you think?

Comment (JSAbrahams) [reply modules/mining-pipeline/jimple-evosuite-test-generator/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/testgenerator/jimpleevosuite/ClassGenerator.kt]: @FWDekker I like it 😄

**Pull request merged**